ffcx/codegeneration/evaluatebasis.py

```diff
@@ -413,14 +410,14 @@ def _idx2d(p, q):
     if embedded_degree > 1:
         f2 = L.Symbol("tmp2_%d" % embedded_degree)
         f3 = L.Symbol("tmp3_%d" % embedded_degree)
-        code += [L.VariableDecl("const double", f2, (1.0 - Y[1]) / 2.0)]
+        code += [L.VariableDecl("const double", f2, (1.0 - Y1) / 2.0)]
         code += [L.VariableDecl("const double", f3, f2 * f2)]
     for r in range(1, embedded_degree):
         rr = _idx2d((r + 1), 0)
         ss = _idx2d(r, 0)
         tt = _idx2d((r - 1), 0)
-        A = (2 * r + 1.0) / (r + 1)
-        B = r / (1.0 + r)
+        A = 1 + r / (r + 1)
```

Comment:
It this correct? Shouldn't it be `2 - 1/(r+1)`?

Reply:
I think it is correct, `(2*r +1)/(r+1) = (r + (r+1))/(r+1) = 1 + r/(r+1)`. But maybe I shouldn't touch it. So many of these expressions are rather cryptic.

Reply:
Ah, ok, both expressions are correct. Yes, many of these formulas are hard to follow without some additional material...